### PR TITLE
addded hsilva/tlsreceiver

### PR DESCRIPTION
**Pull request by hecomp**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by nslaughter — COMMENTED** [review]
remember add certpath to validation

File: receiver/tlscheckreceiver/config.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	configtls.TLSCertsClientSettings        `mapstructure:",squash"`
+
+	metadata.MetricsBuilderConfig `mapstructure:",squash"`
+}
+
+// Validate validates the configuration by checking for missing or invalid fields
+func (cfg *Config) Validate() error {
+	var err error
+
```
Comment (nslaughter) [inline receiver/tlscheckreceiver/config.go:30]: remember to add the certpath validation

File: receiver/tlscheckreceiver/documentation.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+| ---- | ----------- | ------ |
+| error.message | Error message recorded during check | Any Str |
+
+### tlscheck.issued_days
+
+Day duration of timespan certificate is issued for.
+
+| Unit | Metric Type | Value Type | Aggregation Temporality | Monotonic |
+| ---- | ----------- | ---------- | ----------------------- | --------- |
+| s | Sum | Int | Cumulative | false |
```
Comment (nslaughter) [inline receiver/tlscheckreceiver/documentation.md:43]: how do we reconcile the unit type of second with the value we show? Is the value 60 * 60 * 24 * days? If that is so then the metric of issued_days shouldn't have the unit type of `s`. Hmmm!

File: receiver/tlscheckreceiver/documentation.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    enabled: false
+```
+
+### tlscheck.days_left
+
+Days until X.509 certificate expiration.
+
+| Unit | Metric Type | Value Type |
+| ---- | ----------- | ---------- |
+| s | Gauge | Int |
```
Comment (nslaughter) [inline receiver/tlscheckreceiver/documentation.md:21]: see my other comment on units - days vs. seconds. It's not part of UCUM, but seems like `d` might work for days.

File: receiver/tlscheckreceiver/factory.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	return receiver.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiver.WithMetrics(createMetricsReceiver, metadata.Stability))
+}
+
+func createDefaultConfig() component.Config {
+	return &Config{
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			CollectionInterval: 10 * time.Second,
```
Comment (nslaughter) [inline receiver/tlscheckreceiver/factory.go:48]: Collection interval should be from config

File: receiver/tlscheckreceiver/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    enabled: true
+    description: Seconds until X.509 certificate expiration.
+    unit: s
+    gauge:
+      value_type: int
+  tlscheck.issued_days:
+    enabled: true
+    description: Day duration of timespan certificate is issued for.
+    unit: s
+    sum:
```
Comment (nslaughter) [inline receiver/tlscheckreceiver/metadata.yaml:34]: I think this is a gauge, like the remaining time metrics. Check it out.

**Comment by dmitryax** [comment]
@hecomp please follow https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components

**Pull request closed without merging**